multimodal/src/autogluon/multimodal/optimization/lit_distiller.py

```diff
@@ -259,11 +292,19 @@ def _compute_loss(
         )
         loss += hard_label_loss * self.hard_label_weight
 
-        soft_label_loss = self._compute_soft_label_loss(
-            student_output=student_output,
-            teacher_output=teacher_output,
-        )
-        loss += soft_label_loss * self.soft_label_weight
+        if self.soft_label_weight > 0:
+            soft_label_loss = self._compute_soft_label_loss(
+                student_output=student_output,
+                teacher_output=teacher_output,
+            )
+            loss += soft_label_loss * self.soft_label_weight
+
+        if self.softmax_regression_weight > 0:
+            softmax_regression_loss = self._compute_softmax_regression_loss(
```

Comment:
We may need a better name. This loss also belongs to the soft label loss. It just reuses teacher's head.

Reply:
This name is used in the paper, but of course we can change it with a better name.